### ISSUE TITLE
depends-on :git does not honour :ref

### ISSUE THREAD

**Issue by wosc** [cask/cask#456]
### tl;dr

Installing a package specified with git+ref fails with an error about a dependency that the package _does not declare_ at the specified commit ref. Please help? :)

### Long version

 I have an emacs-24.5 and want to install magit. The current version on melpa-stable is 2.90, which depends on emacs-25.1, so `(depends-on "magit")` fails with "Dependency magit failed to install: Package 'emacs-25.1' is unavailable". (At first I didn't even get to see that error due to #450; I've worked around this by manually removing all `shut-up` calls from my `~/.cask/cask.el`.)

So I thought, well the magit project has been around for some time, they should have an older version that worked under emacs-24, which they do: [2.13.1](https://github.com/magit/magit/blob/2.13.1/lisp/magit-pkg.el#L3).

Then I was surprised that `(depends-on)` does not allow specifying the version I want (which I somewhat understood when I learned that melpa currently only provides the [latest version of each package](https://github.com/melpa/melpa/issues/4712) -- even more unexpected ;). But no fear, I thought, let's go directly to the upstream version:

```elisp
;; my entire Cask file
(source gnu)
(source melpa-stable)

(depends-on "magit"
   :git "https://github.com/magit/magit"
   :ref "2.13.0"
   :files ("lisp/*.el"))
```

This seems to retrieve the proper 2.13 code into `/tmp/cask/checkout/magit` and `/tmp/cask/packages/magit-20180603.238.tar` -- and then _still_ fails with "Package 'emacs-25.1' is unavailable", which leaves me totally stumped. What am I doing wrong? Is there some more intermediary debug output other than `cask install --debug` I could enable to make sense of this? Thanks for your help!

**Comment by dickmao** [comment]
Could not reproduce with emacs-27.2.

I would avoid `melpa-stable` and `emacs-25`, generally.
